### PR TITLE
Fix output stream operator

### PR DESCRIPTION
**Pull request by jennuine**
# 🦟 Bug fix

Part of https://github.com/ignitionrobotics/sdformat/issues/790

(See section (2) of https://github.com/ignitionrobotics/sdformat/issues/790#issuecomment-1026187429)

## Summary
The output stream would change the value when [calling `precision()`](https://github.com/ignitionrobotics/ign-math/blob/6761d32609c67c63d138feac07086571b9b5ac16/include/ignition/math/Helpers.hh#L475). This causes issues changing the precision of the output stream.

For example,
 
```c++
math::Vector3d test(1.5707963267948966, 0, 0);
std::cout << test;
// prints 1.5708 0 0

std::cout << std::setprecision(std::numeric_limits<double>::digits10) 
              << test; 
// prints 1.5707960000000001 0 0; x should be 1.5707963267949
```

Since there doesn't seem to be a clear reason to why the value is changed and the default precision is already 6, I've deprecated the function that changes the value `appendToStream(std::ostream &_out, T _number, int _precision)` to be replaced with `appendToStream(std::ostream &_out, T _number)`. 

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Updated documentation (as needed)
- [x] Updated migration guide (as needed)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: src/Helpers_TEST.cc [hunk 1]
```diff
@@ -963,21 +967,41 @@ TEST(HelpersTest, AppendToStream)
 {
   std::ostringstream out;
 
-  math::appendToStream(out, 0.12345678, 3);
```
Comment (azeey) [inline src/Helpers_TEST.cc:966]: Can you keep these tests and use `IGN_UTILS_WARN_IGNORE__DEPRECATED_DECLARATION` and `IGN_UTILS_WARN_RESUME__DEPRECATED_DECLARATION`. We can remove the test when we remove the function.

**Review by azeey — COMMENTED** [review]
Looks great! Do you mind adding checks to the `OperatorStreamOut` test of each of these classes to ensure that the precision set on the stream object is used properly? Right now the tests only cover `appendToStream` for individual numbers.

File: src/Helpers_TEST.cc (hunk 1 above)
Comment (jennuine) [reply src/Helpers_TEST.cc:966]: c35a039b2c4b33867d46380784a3ba70b801d1c9

**Comment by jennuine** [comment]
> Do you mind adding checks to the OperatorStreamOut test of each of these classes to ensure that the precision set on the stream object is used properly?

2f6368130530e692568ba6b124e5213a2572b3f6

**Review by azeey — APPROVED** [review]
LGTM with green CI! See https://github.com/ignitionrobotics/ign-math/pull/374#pullrequestreview-875182543 about OrientedBox_TEST.

File: include/ignition/math/Helpers.hh [hunk 2]
```diff
@@ -476,6 +478,22 @@ namespace ignition
       }
     }
 
+    /// \brief Append a number to a stream. Makes sure "-0" is returned as "0".
+    /// \param[out] _out Output stream.
+    /// \param[in] _number Number to append.
+    template<typename T>
+    inline void appendToStream(std::ostream &_out, T _number)
+    {
+      if (std::fpclassify(_number) == FP_ZERO)
```
[the remark this answers is not in the document]
Comment (jennuine) [reply include/ignition/math/Helpers.hh:499]: Thanks! 2560b0b57fba815e6f6b84e5965a91ebb6767188

**Comment by jennuine** [comment]
> LGTM with green CI! See [#374 (review)](https://github.com/ignitionrobotics/ign-math/pull/374#pullrequestreview-875182543) about OrientedBox_TEST.

Thanks! 2560b0b57fba815e6f6b84e5965a91ebb6767188

File: include/ignition/math/Helpers.hh (hunk 2 above)
[the remark this answers is not in the document]
Comment (jennuine) [reply include/ignition/math/Helpers.hh:499]: I deprecated the previous specialization, does this need to be added to `Changelog.md` and/or `Migration.md`?

[the remark this answers is not in the document]
Comment (azeey) [reply include/ignition/math/Helpers.hh:499]: I don't think you need to add the specialization to the Changelog/Migration.md if you have the template itself in there.

**Pull request merged**